osu.Game/Overlays/Music/PlaylistItem.cs

```diff
@@ -114,24 +117,58 @@ private void recreateText()
             });
         }
 
-        protected override bool OnHover(Framework.Input.InputState state)
+        protected override bool OnHover(InputState state)
         {
             handle.FadeIn(fade_duration);
 
             return base.OnHover(state);
         }
 
-        protected override void OnHoverLost(Framework.Input.InputState state)
+        protected override void OnHoverLost(InputState state)
         {
             handle.FadeOut(fade_duration);
         }
 
-        protected override bool OnClick(Framework.Input.InputState state)
+        protected override bool OnClick(InputState state)
         {
             OnSelect?.Invoke(BeatmapSetInfo);
             return true;
         }
 
+        protected override bool OnDragStart(InputState state)
+        {
+            return true;
+        }
+
+        // Maybe render some ghost text
+        protected override bool OnDrag(InputState state)
+        {
+            return true;
+        }
+
+        protected override bool OnDragEnd(InputState state)
+        {
+            int src = (int)Depth;
+            int dst = MathHelper.Clamp((int)(state.Mouse.Position.Y / Height), 0, playlist.Count - 1);

```

Comment:
The calculation here seems to give an off-by-one after scrolling down the list and dragging. Haven't found the cause just yet, but... https://streamable.com/mgjty

Reply:
I can't reproduce this, but I only have maybe 1.5 pages of songs. Does this still happen for you if you only scroll down a little?

Reply:
I think I know the cause of this - items that span multiple lines:
![image](https://user-images.githubusercontent.com/1329837/29806787-150e71c4-8ccc-11e7-95e0-014e26c3da96.png)

Reply:
rip, that breaks everything. Is this a valid usecase? Is there a plan to use ellipses.

Reply:
Don't think so, but I don't think this should be using mouse position anyway in this case. How about doing something like `if (GetContainingInputManager().DraggedDrawable != this) { /* replace dragged drawable */ }`?